chore: add buildpack base images

rokroskar opened the pull request:
Adds an action to build images for the renkulab global environments from our buildpacks.

Project that tests these builds: https://renkulab.io/p/renku-team/test-new-global-images

leafty commented on .github/workflows/build-global-renkulab-images.yml:
```diff
@@ -0,0 +1,53 @@
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3.2.0
```
Not needed

```suggestion
```

olevski commented on global-images/base/requirements.txt:
question: What is the point of `base`? This is a base image that other can use to build their own images I guess? This will add an arbitrary python version. If you use environment.yaml (which can still be empty) you can pick the python version. But this is also something we can improve or change in the future.

rokroskar replied:
I'm hoping to keep it simple(r) by using pip and maybe we can fix the buildpack to let us specify the python version. But yes, I take your point... I can change it to use conda for now.

olevski replied:
I dont have an issue with the name. I am just not sure how do we expect the image to be used. 

As for being able to pick the python version. That is not that simple to fix because it simply comes from the underlying python buildpack. And I am not sure there is a way to specify the version in our buildpack for some cases (i.e. pip) and then use the original buildpack logic for other cases (poetry, conda).

olevski commented on .github/workflows/build-global-renkulab-images.yml:
```diff
@@ -0,0 +1,53 @@
+name: Build global base images
+
+on:
+  push:
+    paths:
+      - "global-images/**"
+      - ".github/workflows/build-global-renkulab-images.yml"
```
issue: since we tag the images with "latest" every time this runs lets run this on pushes to master only, in addition to the restrictions you already have for code changes.

olevski commented on global-images/README.md:
nitpick(non-blocking): IMO if you put the command for building the image in a `Makefile` and then you use the `Makefile` in the github action you can cut down a lot of the docs and you allow anyone to easily build the images without copying and pasting a large command.

The makefile can you use env vars to decide which flavour and frontend to build.

But we can implement this later at some other point. It is really not a problem.

rokroskar replied:
yeah good idea - where should the Makefile live? in the global-images directory?

olevski replied:
Yeah I think it makes sense if it is in the global-images repo.

rokroskar commented:
@olevski @leafty I've addressed the comments: 

- cleaned up the GHA file
- added a Makefile
- tag latest only on merges to main branch
- added caching (we can remove this if it creates too much noise... cuts down build times by ~30 seconds)

leafty reviewed (dismissed):
LGTM
